### PR TITLE
[#7515] feat(core): Improve fromComment in StringIdentifier

### PR DESCRIPTION
**Pull request by senlizishi**
### What changes were proposed in this pull request?

Improve fromComment in StringIdentifier

### Why are the changes needed?

Fix: #7515 

### Does this PR introduce _any_ user-facing change?

no

### How was this patch tested?

Unit tests

### CODE REVIEW

**Comment by senlizishi** [comment]
@justinmclean PTAL

File: core/src/test/java/org/apache/gravitino/TestStringIdentifier.java [hunk 1]
```diff
@@ -163,6 +163,10 @@ public void testGetStringIdFromComment() {
     String comment1 = "This is a comment";
     StringIdentifier stringIdFromComment2 = StringIdentifier.fromComment(comment1);
     Assertions.assertNull(stringIdFromComment2);
+
+    // Test comment contains parentheses but not the Gravitino prefix
+    String comment2 = "A comment (other info)";
+    Assertions.assertNull(StringIdentifier.fromComment(comment2));
   }
```
Comment (jerryshao) [inline core/src/test/java/org/apache/gravitino/TestStringIdentifier.java:178]: Would you please add more edge cases to cover what you change?

Comment (senlizishi) [reply core/src/test/java/org/apache/gravitino/TestStringIdentifier.java:178]: Hi @jerryshao I have adjusted the code logic and added tests to cover my adjustments.

Comment (senlizishi) [reply core/src/test/java/org/apache/gravitino/TestStringIdentifier.java:178]: > Would you please add more edge cases to cover what you change?

Hi @jerryshao I have adjusted the code logic and added tests to cover my adjustments.
